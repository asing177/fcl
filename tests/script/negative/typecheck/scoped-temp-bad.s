transition initial -> terminal;

@initial
foo () {
  if (true) {
    tmp = true;
    tmp = "bla" // bad
  } else {
    tmp = "bla"
  };
  transitionTo(@terminal);
}
