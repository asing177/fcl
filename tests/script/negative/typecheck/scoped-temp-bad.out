Type error at Line 7, Column 11:
   Expected: 'bool' inferred from the assignment to variable 'tmp' on Line 7, Column 11
   But got:  'text' inferred from literal '"bla"' on Line 7, Column 11